tests/ui/sanitize/sanitizer-dataflow-basic.rs

```diff
@@ -0,0 +1,43 @@
+#![allow(non_camel_case_types)]
+// Verifies that labels are propagated through loads and stores.
+//
+//@ needs-sanitizer-dataflow
+//@ run-pass
+//@ compile-flags: -Zsanitizer=dataflow -Zsanitizer-dataflow-abilist=src/llvm-project/compiler-rt/lib/dfsan/libc_ubuntu1404_abilist.txt,src/llvm-project/compiler-rt/lib/dfsan/done_abilist.txt,src/etc/sanitizer-dataflow-abilist-rust-no-core-no-std.txt
+
+use std::mem::size_of;
+use std::os::raw::{c_int, c_long, c_void};
+
+type dfsan_label = u8;
+
+extern "C" {
+    fn dfsan_add_label(label: dfsan_label, addr: *mut c_void, size: usize);
+    fn dfsan_get_label(data: c_long) -> dfsan_label;
+    fn dfsan_has_label(label: dfsan_label, elem: dfsan_label) -> c_int;
+    fn dfsan_read_label(addr: *const c_void, size: usize) -> dfsan_label;
+    fn dfsan_set_label(label: dfsan_label, addr: *mut c_void, size: usize);
+}
+
+pub fn main() {
+    let mut i = 1i64;
+    let i_ptr = &mut i as *mut i64;
+    let i_label: dfsan_label = 1;
+    unsafe {
+        dfsan_set_label(i_label, i_ptr as *mut c_void, size_of::<i64>());
+    }
+
+    let new_label = unsafe { dfsan_get_label(i) };
+    assert_eq!(i_label, new_label);
+
+    let read_label = unsafe { dfsan_read_label(i_ptr as *const c_void, size_of::<i64>()) };
+    assert_eq!(i_label, read_label);
+
+    let j_label: dfsan_label = 2;
+    unsafe {
+        dfsan_add_label(j_label, i_ptr as *mut c_void, size_of::<i64>());
+    }
+
```

Comment:
This could be a separate test...

I suggest send `i` on a trip through some Rust code to confirm the transformation works.

For example: pass it into a function, store it in a vec, add something to it, clone a reference of it, etc. and then confirm that the taint label was propagated through all that.

e.g.
```
fn propagate2(i: &i64) -> i64 {
  i.clone()
}
fn propagate(i: i64) -> i64 {
  let v = vec!(i, 1, 2, 3);
  let j = v.iter().sum();
  propagate2(&j)
}
```

```
  let mut x = propagate(i);
  let x_ptr = &mut x as *mut i64;
  let x_read_label = unsafe { dfsan_read_label(x_ptr as *const c_void, size_of::<i64>()) };
  assert_eq!(unsafe { dfsan_has_label(x_read_label, i_label) }, 1);
  assert_eq!(unsafe { dfsan_has_label(x_read_label, j_label) }, 1);
```

Reply:
Done. I just replicated the basic functionality test from the DFSan tests in the LLVM Project for now. Do you think we should block this until we review and refactor this (and possibly other tests) or is it okay we do it later in a separate PR?

Reply:
LGTM

Reply:
I think that can be later.